python/pyspark/sql/connect/dataframe.py

```diff
@@ -1147,18 +1147,28 @@ def collect(self) -> List[Row]:
         table = self._session.client.to_table(query)
 
         rows: List[Row] = []
-        for row in table.to_pylist():
-            _dict: Dict[Any, Any] = {}
-            for k, v in row.items():
+        i = 0
+        while i < table.num_rows:
+            keys: List[Any] = []
```

Comment:
keys are list of columns names,  can reuse it across different rows?

Reply:
OK.